### PR TITLE
Refactoring and cleaning

### PR DESCRIPTION
**Pull request by Kecz**

### CODE REVIEW

File: backend/tests/api/test_category.py [hunk 1]
```diff
@@ -160,8 +154,5 @@ def test_put_reset(self, client):
 
 class TestCategoryData:
 
-    # TODO write tests for data
     def test(self):
         pass
```
Comment (Dumbldore) [inline backend/tests/api/test_category.py]: Write tests

File: backend/tests/api/test_dataset.py [hunk 2]
```diff
@@ -1,37 +1,39 @@
...
-        url = '/api/dataset/'
+        url = "/api/dataset/"
         resp = client.get(url)
         resp = json.loads(resp.data)
-        for el in resp:
-            if el['name'] == 'test_inria':
-                return el['id']
+        matched_elem = next((el for el in resp if el["name"] == "test_inria"), None)
+        if matched_elem:
+            return matched_elem["id"]
```
Comment (Dumbldore) [inline backend/tests/api/test_dataset.py:16]: What's the point here?

File: backend/workers/lib/vod_converter/detrac.py [hunk 3]
```diff
@@ -10,66 +10,66 @@
...
                 tree = ET.parse(lab.path)
                 toor = tree.getroot()
-                accepted_tags = ['car', 'bus', 'van']
+                accepted_tags = ["car", "bus", "van"]
                 obj_id = -1
-                for frame in toor.iter('frame'):                # child = frame
-                    no_frame = frame.attrib['num']
-                    img_name = 'img' + '0' * (5 - len(str(no_frame))) + str(no_frame)
-                    for target in frame:                 
+                for frame in toor.iter("frame"):  # child = frame
```
Comment (Dumbldore) [inline backend/workers/lib/vod_converter/detrac.py]: what is this comment?

**Review by Dumbldore — COMMENTED** [review]
some small fixes needed

File: backend/workers/lib/vod_converter/split_labels_from_json_string.py [hunk 4]
```diff
@@ -14,30 +14,28 @@ def split_coco_labels(json_string, max_byte_size):
...
-
         splitted_labels.append(json.dumps(
             {"images": image_chunks[i],
-             'categories': json_dict['categories'],
-             'annotations': [annotation for img in image_chunks[i] for annotation in annotations_base[img['id']]]}))
+             "categories": json_dict["categories"],
+             "annotations": [annotation for img in image_chunks[i] for annotation in annotations_base[img["id"]]]}))
         print(f"Created JSON substring nr {i}")
 
     return splitted_labels
```
Comment (korjak) [inline backend/workers/lib/vod_converter/split_labels_from_json_string.py:41]: is that file even used in annotator?

File: backend/workers/lib/vod_converter/udacity.py [hunk 5]
```diff
@@ -1,146 +0,0 @@
...
-import glob
-import os
-from collections import defaultdict
-
-from PIL import Image
-
-from .abstract import Ingestor
-
-
-class UdacityCrowdAIIngestor(Ingestor):
```
Comment (korjak) [inline backend/workers/lib/vod_converter/udacity.py:17]: why delete this file?

File: backend/workers/tasks/data.py [hunk 6]
```diff
@@ -344,19 +309,18 @@ def import_annotations(task_id, dataset_id, encoded_coco_json):
         ).first()
 
         if annotation_model is None:
-            task.info(f"Creating annotation data ({image_id}, {category_id}), loaded {annotation_number}/{annotation_list} annotations")
+            task.info(f"Creating annotation data ({image_id}, {category_id}) "
+                      f"({annotation_counter+1}/{total_annotations})")
```
Comment (korjak) [inline backend/workers/tasks/data.py:313]: is that correct syntax?

File: backend/workers/tasks/data.py [hunk 7]
```diff
@@ -366,7 +330,8 @@ def import_annotations(task_id, dataset_id, encoded_coco_json):
             image_categories.append(category_id)
         else:
             annotation_model.update(deleted=False, isbbox=isbbox)
-            task.info(f"Annotation already exists (i:{image_id}, c:{category_id})")
+            task.info(f"Annotation already exists (i:{image_id}, c:{category_id}) "
+                      f"({annotation_counter+1}/{total_annotations})")
```
Comment (korjak) [inline backend/workers/tasks/data.py:334]: questioning syntax again

File: backend/tests/cred.json [hunk 8]
```diff
@@ -1 +0,0 @@
-{"username": "korjak", "password": "korjak"}
```
Comment (korjak) [inline backend/tests/cred.json:1]: credentials are required to complete all tests, thus this file cannot be deleted

**Review by korjak — CHANGES_REQUESTED** [review]
Few fixes required, the most crucial is re-adding the credentials file

File: backend/tests/api/test_dataset.py (hunk 2 above)
Comment (korjak) [reply backend/tests/api/test_dataset.py:16]: > What's the point here?

what do you mean?

File: backend/workers/lib/vod_converter/split_labels_from_json_string.py (hunk 4 above)
Comment (Kecz) [reply backend/workers/lib/vod_converter/split_labels_from_json_string.py:41]: yes, it is used  to split to huge coco jsons

File: backend/workers/lib/vod_converter/udacity.py (hunk 5 above)
Comment (Kecz) [reply backend/workers/lib/vod_converter/udacity.py:17]: It is not used now, it may be restored if needed

File: backend/tests/cred.json (hunk 8 above)
Comment (Kecz) [reply backend/tests/cred.json:1]: Ok, reverting this file with default data

**Review by Dumbldore — APPROVED** [review]

**Review by korjak — APPROVED** [review]

**Pull request merged**